components/sendgrid/actions/list-blocks/list-blocks.js

```diff
@@ -0,0 +1,40 @@
+const sendgrid = require("../../sendgrid.app");
+
+module.exports = {
+  key: "sendgrid-list-blocks",
+  name: "List Blocks",
+  description: "Allows you to list all email addresses that are currently on your blocks list.",
+  version: "0.0.6",
+  type: "action",
+  props: {
+    sendgrid,
+    startTime: {
+      type: "integer",
+      label: "Start Time",
+      description: "The start of the time range when a blocked email was created (inclusive). This is a unix timestamp.",
+      optional: true,
+    },
+    endTime: {
+      type: "integer",
+      label: "End Time",
+      description: "The end of the time range when a blocked email was created (inclusive). This is a unix timestamp.",
+      optional: true,
+    },
+    limit: {
+      type: "integer",
+      label: "Limit",
+      description:
+        "Limit the number of results to be displayed per page.",
+      optional: true,
+    },
+    offset: {
+      type: "integer",
+      label: "Offset",
+      description: "The point in the list to begin displaying results.",
+      optional: true,
+    }
```

Comment:
Similar to my comments above regarding pagination, these props should not be exposed to the user since they're out of their control. We should instead accept a parameter like _Number of blocked emails to return_:
![image](https://user-images.githubusercontent.com/5479513/121625998-6c0a0d00-ca29-11eb-8741-6cf66bf3d50e.png)